packages/domain/lib/use_case/request_movie_list_use_case.dart

```diff
@@ -13,33 +17,71 @@ enum TypeListMovie {
 class RequestMovieListUseCase
     extends UseCaseInOut<TypeListMovie, Future<List<MovieListResponse>>> {
   final NetworkTraktRepository _networkRepository;
+  final DatabaseRepository _databaseRepository;
+  final PreferencesLocalRepository _preferencesLocalRepository;
+  final IsTodayDateMapper isTodayDateMapper;
+  final GetLoadDateMapper getLoadDateMapper;
 
   RequestMovieListUseCase(
     this._networkRepository,
+    this._databaseRepository,
+    this._preferencesLocalRepository,
+    this.isTodayDateMapper,
+    this.getLoadDateMapper,
   );
 
   @override
   Future<List<MovieListResponse>> call(TypeListMovie params) async {
+    final dateLoad = params == TypeListMovie.trending
+        ? _preferencesLocalRepository.getDateLoadTrendingMovieList()
+        : _preferencesLocalRepository.getDateLoadComingMovieList();
+    if (isTodayDateMapper(dateLoad)) {
+      return await _databaseRepository.readMovieList(params);
+    } else {
+      final List<MovieListResponse> movieList = [];
+      await _requestListMovie(
+        movieList: movieList,
+        countItem: await _getCountItemAndSaveDate(params),
+        params: params,
+      );
+      await _compareListWithDb(
+        movieList,
+        params,
+      );
+      return movieList;
+    }
+  }
+
+  Future<void> _compareListWithDb(
+    List<MovieListResponse> movieList,
+    TypeListMovie params,
+  ) async {
+    final newListId = movieList.map((e) => e.movie?.ids?.trakt).toList();
+    if (await _databaseRepository.isEqualToMovieListWithDb(newListId, params)) {
+      await _databaseRepository.deleteMovieList(params);
+      await _databaseRepository.insertMovieList(movieList, params);
```

Comment:
don't delete all movies. delete only the movies u don't need. and the same logic with insert